django/db/models/base.py

```diff
@@ -878,7 +878,7 @@ def _get_next_or_previous_in_order(self, is_next):
     def prepare_database_save(self, field):
         if self.pk is None:
             raise ValueError("Unsaved model instance %r cannot be used in an ORM query." % self)
-        return getattr(self, field.remote_field.field_name)
+        return getattr(self, field.remote_field.get_attname())
```

Comment:
Got something working. We might change this to `field.remote_field.get_related_field().attname` and omit the new method if you think that's better.